addons/mrp/models/mrp_production.py

```diff
@@ -532,6 +532,22 @@ def _workorders_create(self, bom, bom_data):
             workorder._generate_lot_ids()
         return workorders
 
+    def _check_lots(self):
+        # Check that the raw materials were consumed for lots that we have produced.
+        if self.product_id.tracking != 'none':
+            finished_lots = set(self.finished_move_line_ids.mapped('lot_id'))
+            raw_finished_lots = set(self.move_raw_ids.mapped('move_line_ids.lot_produced_id'))
+            if not (raw_finished_lots <= finished_lots):
+                lots_short = raw_finished_lots - finished_lots
+                error_msg = _(
+                    'Some raw materials were produced for a lot without finished product. '
+                    'You can correct the following components by unlocking:\n'
+                )
+                move_lines = self.move_raw_ids.mapped('move_line_ids').filtered(lambda x: x.lot_produced_id in lots_short)
+                for ml in move_lines:
+                    error_msg += ml.product_id.name + ' (' + ml.lot_produced_id.name +')\n'
```

Comment:
@nim-odoo could you use `product_id.display_name` than `product_id.name` if case it is a variante. Thx